src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -1351,6 +1352,24 @@ private void ParseExtensibleDirective(in SyntaxListBuilder<RazorSyntaxNode> buil
                                     builder.Add(BuildDirective());
                                     return;
                                 }
+                                break;
+                            case DirectiveTokenKind.Attribute:
+                                if (At(SyntaxKind.LeftBracket))
+                                {
+                                    if (Balance(directiveBuilder, BalancingModes.NoErrorOnFailure | BalancingModes.StopAtEndOfLine))
```

Comment:
So will this break if you split up an attribute across lines?
```
@attribute [Foo(@"
 Blazor is cool!

")]
```

Reply:
Oh goodness, multiline attributes are definitely a thing but make me cringe. @ajaybhargavb out of curiosity do we support this for things like `@model` or `@inject` with complex types?

Reply:
Uhh yeah. We do support things like,
```
@model Dictionary<string,
    string>
```
I'll remove the newline restriction.